src/xcode/ENA/ENA/Source/Services/TicketValidation/AccessToken/TicketValidationAccessTokenProcessor.swift

```diff
@@ -0,0 +1,85 @@
+//
+// 🦠 Corona-Warn-App
+//
+
+import Foundation
+import ENASecurity
+import SwiftJWT
+
+struct TicketValidationAccessTokenProcessor {
+
+	// MARK: - Init
+
+	init(jwtVerification: JWTVerifying) {
+		self.jwtVerification = jwtVerification
+	}
+
+	// MARK: - Internal
+	
+	func process(
+		jwtWithHeadersModel: JWTWithHeadersModel,
+		accessTokenSignJwkSet: [JSONWebKey],
+		completion: @escaping (Result<TicketValidationAccessTokenResult, TicketValidationAccessTokenProcessingError>) -> Void
+	) {
+		guard let nonceBase64 = jwtWithHeadersModel.headers["x-nonce"] as? String else {
+			Log.error("Missing header field x-nonce", log: .ticketValidation)
+			completion(.failure(.UNKNOWN))
+			return
+		}
+
+		let accessToken = jwtWithHeadersModel.jwt
+
+		switch jwtVerification.verify(jwtString: accessToken, against: accessTokenSignJwkSet) {
+		case .success:
+			guard let jwtObject = try? JWT<TicketValidationAccessToken>(jwtString: accessToken) else {
+				completion(.failure(.ATR_PARSE_ERR))
+				return
+			}
+
+			let accessTokenPayload = jwtObject.claims
+
+			guard accessTokenPayload.t == 1 || accessTokenPayload.t == 2 else {
```

Comment:
Would be nice to have the tech-spec entries as comments here for the different steps.